url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/com/mxgraph/view/mxGraph.java

```diff
@@ -0,0 +1,7565 @@
+	 * the selection cells.
+	 */
+	public Object[] toggleCellStyleFlags(String key, int flag, Object[] cells)
+	{
+		return setCellStyleFlags(key, flag, null, cells);
+	}
+
+	/**
+	 * Sets or toggles the given bit for the given key in the styles of the
+	 * selection cells.
+	 * 
+	 * @param key String representing the key to toggle the flag in.
+	 * @param flag Integer that represents the bit to be toggled.
+	 * @param value Boolean value to be used or null if the value should be
+	 * toggled.
+	 */
+	public Object[] setCellStyleFlags(String key, int flag, boolean value)
+	{
+		return setCellStyleFlags(key, flag, value, null);
+	}
+
+	/**
+	 * Sets or toggles the given bit for the given key in the styles of the
+	 * specified cells.
+	 * 
+	 * @param key String representing the key to toggle the flag in.
+	 * @param flag Integer that represents the bit to be toggled.
+	 * @param value Boolean value to be used or null if the value should be
+	 * toggled.
+	 * @param cells Optional array of cells to change the style for. If no
+	 * cells are specified then the selection cells are used.
+	 */
+	public Object[] setCellStyleFlags(String key, int flag, Boolean value,
+			Object[] cells)
+	{
+		if (cells == null)
+		{
+			cells = getSelectionCells();
+		}
+
+		if (cells != null && cells.length > 0)
+		{
+			if (value == null)
+			{
+				mxCellState state = view.getState(cells[0]);
+				Map<String, Object> style = (state != null) ? state.getStyle()
+						: getCellStyle(cells[0]);
+
+				if (style != null)
+				{
+					int current = mxUtils.getInt(style, key);
+					value = !((current & flag) == flag);
+				}
+			}
+
+			mxUtils.setCellStyleFlags(model, cells, key, flag, value);
+		}
+
+		return cells;
+	}
+
+	//
+	// Cell alignment and orientation
+	//
+
+	/**
+	 * Aligns the selection cells vertically or horizontally according to the
+	 * given alignment.
+	 * 
+	 * @param align Specifies the alignment. Possible values are all constants
+	 * in mxConstants with an ALIGN prefix.
+	 */
+	public Object[] alignCells(String align)
+	{
+		return alignCells(align, null);
+	}
+
+	/**
+	 * Aligns the given cells vertically or horizontally according to the given
+	 * alignment.
+	 * 
+	 * @param align Specifies the alignment. Possible values are all constants
+	 * in mxConstants with an ALIGN prefix.
+	 * @param cells Array of cells to be aligned.
+	 */
+	public Object[] alignCells(String align, Object[] cells)
+	{
+		return alignCells(align, cells, null);
+	}
+
+	/**
+	 * Aligns the given cells vertically or horizontally according to the given
+	 * alignment using the optional parameter as the coordinate.
+	 * 
+	 * @param align Specifies the alignment. Possible values are all constants
+	 * in mxConstants with an ALIGN prefix.
+	 * @param cells Array of cells to be aligned.
+	 * @param param Optional coordinate for the alignment.
+	 */
+	public Object[] alignCells(String align, Object[] cells, Object param)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 32. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.